OSEM Installation using Docker doesn't work

**I'm submitting a ..**
- [x] Bug Report
- [ ] Feature Request

**Current behavior:**
<!-- Describe the bug -->
When following INSTALL.md to install osem using docker in openSUSE Leap 15.0, it doesn't work as expected. When I run 
```
docker-compose -f docker-compose.yml.production-example build
```
it produce some errors. The errors are:
```
ost-install message from rails-assets-markdown:
This component doesn't define main assets in bower.json.
Please open new pull request in component's repository:
https://github.com/evilstreak/markdown-js
Removing intermediate container b64c91e84b75
 ---> 457949717fa5
Step 6/8 : RUN export RAILS_ENV=production; bundle exec rake assets:clobber assets:precompile
 ---> Running in d17ce16eb04a
rake aborted!
LoadError: cannot load such file -- rubocop/rake_task
/osem/lib/tasks/rubocop.rake:3:in `<top (required)>'
/osem/Rakefile:9:in `<top (required)>'
(See full trace by running task with --trace)
ERROR: Service 'production_web' failed to build: The command '/bin/sh -c export RAILS_ENV=production; bundle exec rake assets:clobber assets:precompile' returned a non-zero code: 1
```

**Expected correct behavior:**
<!-- Describe what should happen, without the bug. -->
Images are build.

**Steps to reproduce:**
<!-- If you are able to illustrate the bug or feature request with an example, please provide steps to reproduce it. -->
In openSUSE Leap 15.0 follow the step in INSTALL.md and run 
```
docker-compose -f docker-compose.yml.production-example build
```

**Other information:**
<!-- List any other information that is relevant to your issue. Stack traces, related issues, suggestions on how to fix, Stack Overflow links, forum links, etc. -->
I can install it by adding tmp directory in osem directory and give it sticky bit.
```
cd osem
mkdir tmp
chmod 777 tmp/
chmod o+t tmp/
```
Then running 
```
docker-compose up -d
```
The container is up and running and I can access it through browser

![Screenshot_20190502_190524](https://user-images.githubusercontent.com/1616257/57074204-6cddfa00-6d0d-11e9-8492-f7b71460cb21.png)

Without adding tmp directory with sticky bit in osem directory there will be a problem with the permission when trying to acces it using browser. Below is the example
![Screenshot_20190426_183850](https://user-images.githubusercontent.com/1616257/57074022-09ec6300-6d0d-11e9-87dc-9a81bf6c7908.png)

I'm having the same issue, though with the current version of the Dockerfile.production and the tmp file commands in osem directory don't seem to fix it.

```
Step 5/8 : RUN export NOKOGIRI_USE_SYSTEM_LIBRARIES=1; bundle install --system --jobs=3 --retry=3 --without test development
 ---> Running in 5b75dbe14359
Following files may not be writable, so sudo is needed:
  /usr/bin
  /usr/lib64/ruby/gems/2.5.0
  /usr/lib64/ruby/gems/2.5.0/build_info
  /usr/lib64/ruby/gems/2.5.0/cache
  /usr/lib64/ruby/gems/2.5.0/doc
  /usr/lib64/ruby/gems/2.5.0/extensions
  /usr/lib64/ruby/gems/2.5.0/gems
  /usr/lib64/ruby/gems/2.5.0/specifications
Your Ruby version is 2.5.5, but your Gemfile specified 2.5.0
ERROR: Service 'production_web' failed to build: The command '/bin/sh -c export NOKOGIRI_USE_SYSTEM_LIBRARIES=1; bundle install --jobs=3 --retry=3 --without test development' returned a non-zero code: 18
make: *** [Makefile:3: build] Error 1
```
note that neither the "ruby version" or the "files may not be writable" messages seem to be the cause of the problem, as either setting an OSEM_RUBY_VERSION in .env.production or running bundle as sudo ... --system makes the respective warnings disappear, but the error stays the same.
I've not been able to interpret the error code.
I did find similar issue somewhere (exit code 18) that said this error was introduced by some update around ruby 2.5.3. I've been trying to downgrade the ruby version to 2.5.0 (replacing the zypper install with building rvm followed by installing a specific ruby version within it) in the Dockerfile.base image since midnight to no avail so far.

--- a/Dockerfile.production
+++ b/Dockerfile.production
@@ -1,20 +1,28 @@
 FROM registry.opensuse.org/opensuse/infrastructure/dale/containers/osem/base:latest
 
+EXPOSE 5000
+
 ENV RAILS_ENV=production
+ENV RAILS_SERVE_STATIC_FILES=true
 
 # Add our files
 COPY --chown=1000:1000 . /osem/
 
-# Install bundler & foreman
-RUN gem install bundler -v "$(grep -A 1 "BUNDLED WITH" /osem/Gemfile.lock | tail -n 1)"; \
-    gem install foreman
+# Install foreman
+RUN gem install foreman
 
 USER osem
 WORKDIR /osem/
 
+RUN rm -rf /osem/public/system/sponsors; mkdir -p /osem/public/system/sponsors; \
+    rm -rf /osem/public/assets; mkdir -p /osem/public/assets; \
+    rm -rf /osem/log; mkdir -p /osem/public/assets
+
 # Install our bundle
-RUN bundle config set --local without 'test development'
-RUN bundle install --jobs=3 --retry=3
+RUN bundle config set --local without 'test development'; \
+    bundle config set --local path 'vendor/bundle'; \
+    bundle config build.nokogiri --use-system-libraries; \
+    bundle install --jobs=4 --retry=3
 
 # Generate assets
 RUN bundle exec rake assets:precompile

--- a/INSTALL.md
+++ b/INSTALL.md
@@ -37,11 +37,11 @@ There are too many to list, here are a few we have seen OSEM deployed to:
 
 We recommend to run OSEM in production with [mod\_passenger](https://www.phusionpassenger.com/download/#open_source)
 and the [apache web-server](https://www.apache.org/). There are tons of guides on how to deploy rails apps on various
-base operating systems. [Check Google](https://encrypted.google.com/search?hl=en&q=ruby%20on%20rails%20apache%20passenger). Of course there are also other options for the application server and reverse proxy, pick your poison.
+base operating systems, [check DuckDuckGo](https://duckduckgo.com/?t=h_&q=ruby+on+rails+apache+passenger). Of course there are also other options for the application server and reverse proxy, pick your poison.
 
 ## Deploy via docker/docker-compose
 
-There is a rudimentary docker-compose configuration for production usage (`docker-compose.yml.production-example`). It brings [OSEM up](http://0.0.0.0:8080) on port 8080. It uses persistent storage volumes for all the data users might create. You can start it with
+There is a rudimentary docker-compose configuration for production usage (`docker-compose.yml.production-example`). It brings OSEM up on port 80. It uses persistent storage volumes for all the data your users might create. You can start it with
 
 1. Configure OSEM (at least `SECRET_KEY_BASE`)
    ```
@@ -54,12 +54,13 @@ There is a rudimentary docker-compose configuration for production usage (`docke
    ```
 1. Setup the database (only once)
    ```
-   docker-compose -f docker-compose.yml.production-example run --rm production_web bundle exec rake db:bootstrap
+   docker-compose -f docker-compose.yml.production-example run --rm osem bundle exec rake db:setup
    ```
 1. Start the services
    ```
    docker-compose -f docker-compose.yml.production-example up
    ```
+1. Visit http://0.0.0.0
 
 ## Configuration
 OSEM is configured through environment variables and falls back to sensible defaults. See the [dotenv.example](https://github.com/openSUSE/osem/blob/master/dotenv.example) for all possible configuration options. However here is a list of things you most likely want to configure because otherwise things will not work as expected.

--- a/docker-compose.yml.production-example
+++ b/docker-compose.yml.production-example
@@ -1,24 +1,25 @@
 services:
-  production_database:
-    image: postgres:12-alpine
+  postgres:
+    image: postgres:16-alpine
     environment:
       PGDATA: /var/lib/postgresql/data/pgdata
+      POSTGRES_USER: osem
       POSTGRES_PASSWORD: mysecretpassword
     volumes:
       - osem_production_database:/var/lib/postgresql/data/pgdata
-  production_web:
+  osem:
     build:
       context: .
       dockerfile: Dockerfile.production
     depends_on:
-      - production_database
+      - postgres
     ports:
-      - 8080:3000
+      - 80:5000
     env_file: .env.production # see dotenv.example file
     environment:
-     OSEM_DB_HOST: production_database
-     RAILS_SERVE_STATIC_FILES: 'true'
-    command: foreman start -p 3000
+     OSEM_DB_HOST: postgres
+     OSEM_DB_USER: osem
+     OSEM_DB_PASSWORD: mysecretpassword
     volumes:
       - osem_production_web_data:/osem/public/system
       - osem_production_web_assets:/osem/public/assets

